timesketch/api/v1/resources/sigma.py

```diff
@@ -153,11 +217,371 @@ def post(self):
+                "Sigma parsing error: invalid yaml provided: {0!s}".format(
+                    exception
+                ),
             )
 
-        if sigma_rule is None:
+        if not sigma_rule:
             abort(HTTP_STATUS_CODE_NOT_FOUND, "No sigma was parsed")
+
+        return jsonify({"objects": [sigma_rule], "meta": {}})
+
+
+class SigmaRuleListResource(resources.ResourceMixin, Resource):
+    """Resource to get list of all SigmaRules."""
+
+    @login_required
+    def get(self):
+        """Fetches Sigma rules from the database
+
+        Fetches fetch all Sigma rules stored in the database on the system
+        and returns a list of JSON representations of the rules.
+
+
+        Returns:
+            List of sigma rules represented in JSON e.g.
+                {"objects": [sigma_rules], "meta": {"rules_count": 42}
+        """
+        sigma_rules = []
+
+        all_sigma_rules = SigmaRule.query.all()
+        for rule in all_sigma_rules:
+            sigma_rules.append(_enrich_sigma_rule_object(rule=rule))
+
+        meta = {"rules_count": len(sigma_rules)}
+        return jsonify({"objects": sigma_rules, "meta": meta})
+
+
+class SigmaRuleResource(resources.ResourceMixin, Resource):
+    """Resource to read / delete / create / update a Sigma rule."""
+
+    @login_required
+    def get(self, rule_uuid):
+        """Fetches a single Sigma rule from the databse.
+
+        Fetches a single Sigma rule selected by the `uuid` in
+        `/sigmarule/<string:rule_uuid>/` and returns a JSON represantion of the
+        rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            JSON sigma rule representation
+            e.g.:  {"objects": [return_rule], "meta": {}}
+        """
+        try:
+            rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        except Exception as e:  # pylint: disable=broad-except
+            error_msg = "Unable to get the Sigma rule {0!s}".format(e)
+            logger.error(
+                error_msg,
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_INTERNAL_SERVER_ERROR,
+                error_msg,
+            )
+
+        if not rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No rule found with this ID.")
+        return_rule = []
+
+        assert isinstance(rule, SigmaRule)
+
+        return_rule.append(_enrich_sigma_rule_object(rule=rule))
+
+        return jsonify({"objects": return_rule, "meta": {}})
+
+    @login_required
+    def delete(self, rule_uuid):
+        """Deletes a Sigma rule from the database,.
+
+        Deletes a single Sigma rule selected by the `uuid` in
+        `/sigmarule/<string:rule_uuid>/` and returns a HTTP Code 200 if
+        the rule is deleted.
+
+        Args:
+            rule_uuid: uuid of the rule to be deleted
+
+        Returns:
+            HTTP_STATUS_CODE_NOT_FOUND if rule not found
+            HTTP_STATUS_CODE_OK if rule deleted
```

Comment:
```suggestion
            HTTP_STATUS_CODE_NOT_FOUND if rule not found.
            HTTP_STATUS_CODE_OK if rule deleted.
```

Please go over the docstrings and add periods at the end of sentences.